packages/icrc-ledger-agent/src/lib.rs

```diff
@@ -262,11 +262,16 @@ impl Icrc1Agent {
         )?)
     }
 
-    pub async fn get_data_certificate(&self) -> Result<DataCertificate, Icrc1AgentError> {
-        Ok(Decode!(
-            &self.query("get_data_certificate", &Encode!()?).await?,
-            DataCertificate
-        )?)
+    pub async fn get_data_certificate(
+        &self,
+    ) -> Result<icrc_ledger_types::icrc3::blocks::ICRC3DataCertificate, Icrc1AgentError> {
+        Decode!(
+            &self.query("icrc3_get_tip_certificate", &Encode!()?).await?,
+            Option<icrc_ledger_types::icrc3::blocks::ICRC3DataCertificate>
+        )?
+        .ok_or(Icrc1AgentError::VerificationFailed(
+            "DataCertificate not found".to_string(),
+        ))
```

Comment:
Is this to maintain backwards compatibility?

Reply:
The function expects a certificate to be issued. The original function returns the Certificate while the icrc3 function returns an Option of the Certificate. This is why, I return an error if the certificate is missing.

Reply:
Ah, I see, thanks for the explanation!

Are there external users of `icrc-ledger-agent` that call `get_data_certificate`? Since the signature now changes (returning a `Result` of `ICRC3DataCertificate` instead of a `DataCertificate`), would it make sense to add the new functionality as a new method, e.g., `icrc3_get_tip_certificate`, alongside the existing `get_data_certificate` (IIUC, we could still have `get_data_certificate` return the legacy `DataCertificate` as before)? Or if not, rename this `get_data_certificate` method to `icrc_get_tip_certificate`?

Reply:
Done